airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -102,8 +95,17 @@ public Optional<Long> createResetConnectionJob(final DestinationConnection desti
       throws IOException {
     final ConfiguredAirbyteCatalog configuredAirbyteCatalog = standardSync.getCatalog();
     configuredAirbyteCatalog.getStreams().forEach(configuredAirbyteStream -> {
-      configuredAirbyteStream.setSyncMode(SyncMode.FULL_REFRESH);
-      configuredAirbyteStream.setDestinationSyncMode(DestinationSyncMode.OVERWRITE);
+      final StreamDescriptor streamDescriptor = new StreamDescriptor()
+          .withName(configuredAirbyteStream.getStream().getName())
+          .withNamespace(configuredAirbyteStream.getStream().getNamespace());
+      if (streamsToReset.contains(streamDescriptor)) {
+        configuredAirbyteStream.setSyncMode(SyncMode.FULL_REFRESH);
+        configuredAirbyteStream.setDestinationSyncMode(DestinationSyncMode.OVERWRITE);
+      } else {
+        // set sync mode to full_refresh/append so that these streams are NOT affected in the destination
+        configuredAirbyteStream.setSyncMode(SyncMode.FULL_REFRESH);
+        configuredAirbyteStream.setDestinationSyncMode(DestinationSyncMode.APPEND);
```

Comment:
I would like the reviewers to confirm if this change makes sense. 

Currently, we set all stream sync modes to full_refresh/overwrite when performing a reset.

Now, I am setting only the streams in the catalog that are present in `streamsToReset` to full_refresh/overwrite. For streams NOT in `streamsToReset`, I am setting them to full_refresh/append, so that we do not accidentally wipe out any data for them in the destination, which I believe would happen if any of them are set to full_refresh/overwrite in the catalog already.

Is full_refresh/append what we want here for the streams that are not being reset?

Reply:
@lmossman Looks like `FULL_REFRESH` is used in either case.  Does it make more sense to just choose the destination sync mode based on the condition to reduce code duplication?

Reply:
Done